### PR TITLE
[terra-spacer] Update docs to use mdx syntax

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
updated docs to use MDX syntax.

### CODE REVIEW

File: packages/terra-spacer/src/terra-dev-site/doc/spacer/Spacer.1.doc.jsx [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-const DocPage = () => (
-  <DocTemplate
-    packageName={name}
-    packageVersion={version}
-    readme={ReadMe}
-    srcPath={`https://github.com/cerner/terra-core/tree/master/packages/${name}`}
-    examples={[
-      {
-        title: 'Spacer',
-        description: 'Spacing default button with a padding value of large+4 and primary button with padding value of large+2',
```
Comment (emilyrohrbough) [inline packages/terra-spacer/src/terra-dev-site/doc/spacer/Spacer.1.doc.jsx]: We should pull this description into the new mdx file

File: packages/terra-spacer/docs/README.md [hunk 2]
```diff
@@ -1,86 +0,0 @@
...
-  <Image src="someImageSource" alt="example image" />
-<Spacer/>
-
-// Large and very large spacing
-
-<Spacer paddingTop="large" paddingBottom="large+1" paddingLeft="large+2" marginLeft="large+4" marginRight="large+3">
-  <Image src="someImageSource" alt="example image" />
-<Spacer/>
-
-// Shorthand consumption
```
Comment (emilyrohrbough) [inline packages/terra-spacer/docs/README.md]: Do we need to add an example demonstrating short hand vs long hand consumption?

**Comment by avinashg1994** [comment]
The value of all enums are shown as _[ "...Object.values(SpacerSizes)"]_. I think this should be changed to show strings as per this [discussion](https://github.com/cerner/terra-framework/pull/663#discussion_r291202091).

File: packages/terra-spacer/docs/README.md (hunk 2 above)
Comment (pranav300) [reply packages/terra-spacer/docs/README.md]: done here https://github.com/cerner/terra-core/pull/2872/commits/4efa60a5a6bd274e0e0926452d27d02ba78d8ad7

File: packages/terra-spacer/src/terra-dev-site/doc/spacer/Spacer.1.doc.jsx (hunk 1 above)
Comment (pranav300) [reply packages/terra-spacer/src/terra-dev-site/doc/spacer/Spacer.1.doc.jsx]: done here https://github.com/cerner/terra-core/pull/2872/commits/4efa60a5a6bd274e0e0926452d27d02ba78d8ad7

**Pull request merged**